packages/nimble-components/src/theme-provider/design-token-comments.ts

```diff
@@ -9,6 +9,22 @@ export const comments: { readonly [key in TokenName]: string | null } = {
     dividerBackgroundColor: 'Divider background color',
     headerBackgroundColor: 'Background color for application headers',
     sectionBackgroundColor: 'Background color for sections',
```

Comment:
I'm trying to decide if I like these token names and whether they conform to our [token naming guidelines](https://github.com/ni/nimble/blob/123c631d25df675963ee5010eed7ebd203bd2898/packages/nimble-components/CONTRIBUTING.md#token-naming). My hesitation comes from not knowing where the variant parts like "Primary" or "BlockAccent" should go in the name. 

1. Is it a "state" in the guidelines, in which case it should go later:
`buttonBackgroundPrimaryColor`

2. Or is it a new concept, in which case we should describe it in the naming guidelines? I'd be ok with the guidelines saying that variants, if needed, go between "element" and "part".

Could you scan the existing token names to see if there's any precedence? If not we can pick whatever people like better (my vote is 2).

Reply:
Also, what does "Action" mean in these names?

Reply:
Action referred to an action like clicking the button, as the tokens with "Action" in them were used in the .control:active css class. Looking back at it, this might be a bit confusing, so maybe a name like buttonPrimaryClickColor / buttonPrimaryFillClickColor would work better?

Reply:
And in a couple cases there are multiple variant words in one token, e.g. `buttonOutlineAccentBorderColor`. It's a little confusing because "outline" sounds like it might be a part, but it's not. "Border" is the part.

Reply:
The links provide strong precedence and (pretty good) alignment to the conventions:
 
element: link
part: (n/a)
states: Active, Disabled, Prominent
types: Font*

The only one that messed up is linkProminentDisabledFontColor should be linkDisabledProminentFontColor as the states should be in alphabetical order

https://github.com/ni/nimble/blob/17bef3c5becd15e8c25c735b42ba255f134a4e32/packages/nimble-components/src/theme-provider/design-token-comments.ts#L152-L185

I think it helps to write it down, your proposal is combinations of:

element: button
parts: fill, border, ~~background~~ (I think fill and backgrounds are synonyms for the part of the control, we are unfortunately inconsistent but I think we should settle on fill),
states: Primary, Accent, ~~"Action"~~ Active (active is the state the button is in where these are applied), Block, Outline

My questions:
- are the fills completely unique to the button only? Could they be used in other controls as `fillPrimaryColor`, `fillActivePrimaryColor`, `fillAccentColor`, `fillAccentActiveColor` instead?
- is the border color button specific? Could it be `borderAccentColor` instead?

Reply:
I asked Brandon about this, once he responds I'll report what he says here 👍

Reply:
@rajsite I talked to Brandon about this, and he believes that the appearance variant color design tokens will only be used for the button. The one exception is the design token for the accent outline border color, which is used for the link rollover. Although, Brandon believes that using one token for both could be problematic if he eventually needs to make a change to the link colors, and the naming would be unusual since in the button case it's used for a border, but in the link use case it's used for text color.

For the appearance variant design tokens in general, he said that the colors he chose were very specific- he said that he doesn't plan on using combinations like Black91 with 75% or 85% opacity or Black15 with 20% opacity anywhere else and usually sticks to 10%, 30%, and 60% opacities for most nimble components. He said that the button variations had specific use cases that prompted him to use other opacities.

Reply:
Thanks for checking! Then I think rewriting the names to follow the conventions with the breakdown shown below makes sense (assuming I didn't miss anything :P):

element: button
parts: fill, border,
states: Primary, Accent, Active, Block, Outline